main.cpp

```diff
@@ -491,31 +491,62 @@ void preprocesscustomizationfile(std::vector<Student*> &students) {
     assert (gradeable_total_percent >= 0);
     sum_of_percents += gradeable_total_percent;
 
-    int count = one_gradeable_type.value("count",-1);
+    int parse_count = one_gradeable_type.value("count",-1);
```

Comment:
Adding an extra variable here is kind of gross, but so is parsing with a default value of -1 just so we can clobber that -1 (after the if block on L503 it is expected that the old "count" variable was always >=0). I'd rather avoid future cases where we might escape/get around this logic block and retain a -1 in signed/unsigned comparison, because that's going to lead to some very out of bounds looping.